server/ai_session.go

```diff
@@ -272,20 +296,31 @@ func (sel *AISessionSelector) Remove(sess *AISession) {
 }
 
 func (sel *AISessionSelector) Refresh(ctx context.Context) error {
+	// If we try to add new sessions to the pool the suspender
+	// should treat this as a refresh
+	sel.suspender.signalRefresh()
+
 	sessions, err := sel.getSessions(ctx)
 	if err != nil {
 		return err
 	}
 
 	var warmSessions []*BroadcastSession
 	var coldSessions []*BroadcastSession
+
 	for _, sess := range sessions {
 		// If the constraints are missing for this capability skip this session
 		constraints, ok := sess.OrchestratorInfo.Capabilities.Constraints.PerCapability[uint32(sel.cap)]
 		if !ok {
 			continue
 		}
 
+		// this should not be needed, the GetOrchestrators checks for suspension but was seeing orchestrators get back into
```

Comment:
@ad-astra-video before merging can you add the reason you found concisely to the Todo so developers know why it it's here 🙏.

Reply:
Updated